src/app/zap-templates/templates/app/cluster-objects.zapt

```diff
@@ -48,6 +55,11 @@ namespace {{asUpperCamelCase label}} {
 
 {{#zcl_clusters}}
 namespace {{asUpperCamelCase name}} {
+{{#zcl_enums}}
+{{#if has_more_than_one_cluster}}
+using {{asUpperCamelCase name}} = Clusters::detail::Enums::{{asUpperCamelCase name}};
```

Comment:
Why do we need this here?  Should already be declared in that namespace in cluster-enums.h, no?  And this file includes cluster-enums.h....

Reply:
No, the change here is that the enum is now declared in a global namespace Clusters::detail::Enums when it is shared across structs. So it doesn't exist in this namespace.

Reply:
What I see is that cluster-enums.zapt has:

1. A declaration of the enum in `Clusters::detail::Enums` when it's shared.
2. A `using {{asUpperCamelCase name}} = Clusters::detail::Enums::{{asUpperCamelCase name}};` in the cluster namespace, when it's shared.

This bit here looks like a copy of item 2, but again: why is it needed?

This, again, might have been a lot easier to review and see what's going on if it were just the ZAP change and the resulting changes to existing codegen.  But concretely, compare what's happening to LevelValueEnum in the TotalVolatileOrganicCompoundsConcentrationMeasurement namespace in the generated cluster-objects.h and the generated cluster-enums.h